### PR TITLE
Feature/app ts refactor

### PR DESCRIPTION
**Pull request by adrianmroz**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -57,6 +57,7 @@
     "lint:sass": "sass-lint -v"
   },
   "dependencies": {
+    "@sentry/browser": "4.6.3",
```
Comment (adrianmroz) [inline package.json]: not so fast!

File: src/client/applications/turnilo-application/turnilo-application.tsx [hunk 2]
```diff
@@ -34,18 +34,17 @@ import { replaceHash } from "../../utils/url/url";
 import { CubeView } from "../../views/cube-view/cube-view";
 import { HomeView } from "../../views/home-view/home-view";
 import { NoDataView } from "../../views/no-data-view/no-data-view";
-import "./swiv-application.scss";
+import "./turnilo-application.scss";
 
-export interface SwivApplicationProps {
+export interface TurniloApplicationProps {
   version: string;
   user?: User;
```
Comment (mkuthan) [inline src/client/applications/turnilo-application/turnilo-application.tsx]: Perhaps is not needed but I'm not fully sure. Is it possible to set user from requestdecorator https://github.com/allegro/turnilo/blob/master/docs/security.md ?

File: src/server/routes/mkurl/mkurl.ts [hunk 3]
```diff
@@ -15,73 +15,69 @@
...
+      res.status(400).send({ error: "viewDefinition must be an object" });
+      return;
+    }
 
-      let essence: Essence;
+    let settings: AppSettings;
+    try {
+      settings = await settingsGetter(<GetSettingsOptions> { dataCubeOfInterest: dataCubeName });
+    } catch (e) {
+      res.status(400).send({ error: "Couldn't load settings" });
```
Comment (mkuthan) [inline src/server/routes/mkurl/mkurl.ts]: Original exception is lost

Comment (adrianmroz) [reply src/server/routes/mkurl/mkurl.ts]: As was before. Do we wan't to send this exception to client? Rather not. So maybe just log something?
For what is worth, exception here would be thrown in some weird circumstances, probably crashing app before this request handling. We don't throw there any exception explicitly.

File: src/client/applications/turnilo-application/turnilo-application.tsx (hunk 2 above)
Comment (adrianmroz) [reply src/client/applications/turnilo-application/turnilo-application.tsx]: But for this case it shouldn't be done on server side? Right now `user` is not really used.

File: src/server/routes/mkurl/mkurl.ts [hunk 4]
```diff
@@ -15,73 +15,69 @@
...
-import { SwivRequest } from "../../utils/general/general";
-import { GetSettingsOptions } from "../../utils/settings-manager/settings-manager";
+import { GetSettingsOptions, SettingsGetter } from "../../utils/settings-manager/settings-manager";
 
-const router = Router();
+export function mkurlRouter(settingsGetter: SettingsGetter) {
 
-router.post("/", (req: SwivRequest, res: Response) => {
-  const { dataCubeName, viewDefinitionVersion, viewDefinition } = req.body;
+  let router = Router();
```
Comment (piter75) [inline src/server/routes/mkurl/mkurl.ts]: any specific reason to make it mutable?

File: src/server/routes/plyql/plyql.ts [hunk 5]
```diff
@@ -30,76 +27,72 @@ interface PlyqlOutputFunctions {
...
-    res.status(400).send(errmsg);
-    return;
-  }
-
-  parsedQuery = parsedQuery.substitute(ex => {
-    if (ex instanceof RefExpression && ex.name === dataCube) {
-      return $("main");
+export function plyqlRouter(settingsGetter: SettingsGetter) {
+
+  let router = Router();
```
Comment (piter75) [inline src/server/routes/plyql/plyql.ts]: should it stay mutable?

File: src/server/routes/plywood/plywood.ts [hunk 6]
```diff
@@ -16,92 +16,90 @@
...
+import { Dataset, Expression } from "plywood";
+import { SettingsGetter } from "../../utils/settings-manager/settings-manager";
 
-let router = Router();
+export function plywoodRouter(getSettings: SettingsGetter) {
 
-router.post("/", (req: SwivRequest, res: Response) => {
-  const { dataSource, expression, timezone } = req.body;
-  const dataCube = req.body.dataCube || dataSource; // back compat
+  let router = Router();
```
Comment (piter75) [inline src/server/routes/plywood/plywood.ts]: should it stay mutable?

File: src/server/routes/readiness/readiness.ts [hunk 7]
```diff
@@ -54,51 +44,61 @@ interface ClusterHealth {
...
     LOGGER.log(`Unhealthy cluster host: ${host}. Message: ${message}`);
   });
+}
 
-  const isSomeUnhealthy = unhealthyClusters.length > 0;
-  const overallHealthStatus = isSomeUnhealthy ? ClusterHealthStatus.unhealthy : ClusterHealthStatus.healthy;
+export function readinessRouter(getSettings: SettingsGetter) {
 
-  const httpState = statusToHttpStatusMap[overallHealthStatus];
+  let router = Router();
```
Comment (piter75) [inline src/server/routes/readiness/readiness.ts]: should it stay mutable?

File: src/server/routes/shorten/shorten.ts [hunk 8]
```diff
@@ -14,19 +14,22 @@
...
+export function shortenRouter(settingsGetter: SettingsGetter) {
 
-router.get("/", (req: SwivRequest, res: Response) => {
-  const { url } = req.query;
-  req.getSettings()
-    .then(settings => settings.customization.urlShortener)
-    .then(shortener => shortener.shortenerFunction(url, request))
-    .then(shortUrl => res.json({ shortUrl }))
-    .catch(error => {
+  let router = Router();
```
Comment (piter75) [inline src/server/routes/shorten/shorten.ts]: should it stay mutable?

File: src/server/routes/turnilo/turnilo.ts [hunk 9]
```diff
@@ -15,26 +15,29 @@
...
 import { mainLayout } from "../../views";
 
-var router = Router();
+export function turniloRouter(settingsGetter: SettingsGetter, version: string) {
 
-router.get("/", (req: SwivRequest, res: Response, next: Function) => {
-  req.getSettings()
-    .then(appSettings => {
-      var clientSettings = appSettings.toClientSettings();
+  let router = Router();
```
Comment (piter75) [inline src/server/routes/turnilo/turnilo.ts]: should it stay mutable?

**Pull request merged**